hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/util/ObjectParser.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.request.util;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos
+    .OzoneObj.ObjectType;
+import org.apache.hadoop.ozone.security.acl.OzoneObj;
+
+import static org.apache.hadoop.ozone.OzoneConsts.OZONE_URI_DELIMITER;
+
+/**
+ * Utility class to parse {@link OzoneObj#getPath()}.
+ */
+public class ObjectParser {
+
+  private String volume;
+  private String bucket;
+  private String key;
+
+  /**
+   * Parse the path and extract volume, bucket and key names.
+   * @param path
+   */
+  public ObjectParser(String path, ObjectType objectType) throws OMException {
+    Preconditions.checkNotNull(path);
+    String[] tokens = StringUtils.split(path, OZONE_URI_DELIMITER, 3);
+
+    if (objectType == ObjectType.VOLUME && tokens.length == 1) {
```

Comment:
OzoneObj has similar logic to parse the path. Can you check if we can reuse?

Reply:
This additional class is added so that OzoneObj protoBuf conversion can be avoided, and call this to parse the path.

And also I see there is a bug in the OzoneObj, when object type is BUCKET, then still volume can be null. I think we need to address that.

Reply:
Discussed offline, fine with this new class.